src/idl_gen_rust.cpp

```diff
@@ -0,0 +1,1264 @@
+  code += Indent + Indent;
+  code += "self.inner.start_object(";
+  code += NumToString(struct_def.fields.vec.size());
+  code += ");\n";
+  code += Indent + "}\n\n";
+}
+
+// Set the value of a table's field.
+static void BuildFieldOfTable(const StructDef &struct_def,
+                              const FieldDef &field,
+                              const size_t offset,
+                              std::string *code_ptr) {
+  std::string &code = *code_ptr;
+  std::string lname = struct_def.name;
+  std::transform(lname.begin(), lname.end(), lname.begin(), ::tolower);
+  code += Indent;
+  code += "/// Set the value for field `" + field.name + "`.\n";
+  code += Indent;
+  code += "pub fn add_" + field.name;
+  code += "(&mut self, ";
+  code += field.name + ": ";
+  if (!IsScalar(field.value.type.base_type) && (!struct_def.fixed)) {
+    code += "flatbuffers::UOffsetT";
+  } else {
+    code += GenTypeBasic(field.value.type);
+  }
+  code += ") {\n";
+  code += Indent + Indent;
+  code += "self.inner.add_slot_";
+  code += GenMethod(field) + "(";
+  code += NumToString(offset) + ", ";
+  code += field.name;
+  code += ", " + MapConstant(field);
+  code += ")\n";
+  code += Indent + "}\n\n";
+}
+
+// Set the value of one of the members of a table's vector.
+static void BuildVectorOfTable(const FieldDef &field,
+                               std::string *code_ptr) {
+  std::string &code = *code_ptr;
+  code += Indent;
+  code += "/// Initializes bookkeeping for writing a new `";
+  code += field.name + "` vector.\n";
+  code += Indent;
+  code += "pub fn start_";
+  code += field.name;
+  code += "_vector(&mut self, numElems: usize) {\n";
+  code += Indent + Indent;
+  code += "self.inner.start_vector(";
+  auto vector_type = field.value.type.VectorType();
+  auto alignment = InlineAlignment(vector_type);
+  auto elem_size = InlineSize(vector_type);
+  code += NumToString(elem_size);
+  code += ", numElems, " + NumToString(alignment);
+  code += ")\n";
+  code += Indent + "}\n\n";
+}
+
+// Get the offset of the end of a table.
+static void GenEndOfTable(std::string *code_ptr) {
+  std::string &code = *code_ptr;
+  code += Indent;
+  code += "/// Finalize the current object and return the offset.\n";
+  code += Indent;
+  code += "pub fn end";
+  code += "(&mut self) -> flatbuffers::UOffsetT {\n";
+  code += Indent + Indent + "return self.inner.end_object()\n";
+  code += Indent + "}\n\n";
+}
+
+// Get the offset of the end of a table.
+static void GenFinishOnTable(std::string *code_ptr) {
+  std::string &code = *code_ptr;
+  code += Indent;
+  code += "/// Finish the buffer.\n";
+  code += Indent;
+  code += "pub fn finish";
+  code += "(&mut self, r: flatbuffers::UOffsetT) {\n";
+  code += Indent + Indent + "return self.inner.finish(r)\n";
+  code += Indent + "}\n\n";
+}
+
+// Generic builder functions.
+// This table builder wraps an inner flatbuffers:Builder
+// so these are conveniance functions that delegate to the
+// inner builder
```

Comment:
Why does it generate these? Can this not be generic code that is part of the standard classes?